src/pint/bayesian.py

```diff
@@ -0,0 +1,206 @@
+        How to handle phase wrapping. If True, will use the pulse numbers
+        from the toas object while creating :class:`pint.residuals.Residuals`
+        objects. Otherwise will use the nearest integer.
+    prior_info : dict, optional
+        A dict containing the prior information on free parameters. This parameter
+        supersedes any priors present in the model.
+
+    Notes
+    -----
+    > The `prior` attribute of each free parameter in the `model` object should
+      be set to an instance of :class:`pint.models.priors.Prior`.
+
+    > The parameters of BayesianTiming.model will change for every likelihood function
+      call. These parameters in general will not be the best-fit values. Hence, it is NOT
+      a good idea to save it as a par file.
+
+    > Currently, only uniform and normal distributions are supported in prior_info. More
+      general priors should be set directly in the TimingModel object before creating the
+      BayesianTiming object.
+      Here is an example prior_info object:
+        prior_info = {
+            "F0" : {
+                "distr" : "normal",
+                "mu"    : 1,
+                "sigma" : 0.00001
+            },
+            "EFAC1" : {
+                "distr" : "uniform",
+                "pmin"  : 0.5,
+                "pmax"  : 2.0
+            }
+        }
+    """
+
+    def __init__(self, model, toas, use_pulse_numbers=False, prior_info=None):
+        # Make a deep copy to not mess up the original model.
+        self.model = deepcopy(model)
+        self.toas = toas
+
+        self.param_labels = self.model.free_params
+        self.params = [getattr(model, par) for par in self.param_labels]
+        self.nparams = len(self.param_labels)
+
+        if prior_info is not None:
+            for par in prior_info.keys():
+                distr = prior_info[par]["distr"]
+                if distr == "uniform":
+                    pmax, pmin = prior_info[par]["pmax"], prior_info[par]["pmin"]
+                    getattr(self.model, par).prior = Prior(uniform(pmin, pmax - pmin))
+                elif distr == "normal":
+                    mu, sigma = prior_info[par]["mu"], prior_info[par]["sigma"]
+                    getattr(self.model, par).prior = Prior(norm(mu, sigma))
+
+        self._validate_priors()
+
+        self.likelihood_method = self._decide_likelihood_method()
+
+        self.track_mode = "use_pulse_numbers" if use_pulse_numbers else "nearest"
+
+    def _validate_priors(self):
+        for param in self.params:
+            if not hasattr(param, "prior") or param.prior is None:
+                raise AttributeError(f"Prior is not set for parameter {param.name}.")
+            if isinstance(param.prior._rv, UniformUnboundedRV):
+                raise NotImplementedError(
+                    f"Unbounded uniform priors are not supported. (param : {param.name})"
+                )
+
+    def _decide_likelihood_method(self):

```

Comment:
Does this work for both narrowband and Wideband data?

Reply:
It works only works with narrow band now. I haven't tested with wideband data.

Reply:
OK, probably worth putting in the docs (& an issue?) and maybe checking when the toas are fed in?

Reply:
> checking when the toas are fed in?

Yes... I think this is needed. I'll implement this. Maybe throw a NotImplementedError upon encountering wideband TOAs.

Reply:
Implemented.